libr/anal/fcn.c

```diff
@@ -200,9 +242,10 @@ static void _fcn_tree_print_dot(RBNode *n) {
 #endif
 
 // Find RAnalFunction whose addr is equal to addr
-static RAnalFunction *_fcn_tree_find_addr(RBNode *n, ut64 addr) {
+static RAnalFunction *_fcn_addr_tree_find_addr(RAnal *anal, ut64 addr) {
```

Comment:
I think you can remove this function and directly use `r_rbtree_find`

Reply:
Yes but:

1. `r_rbtree_find` requires a `RBComparator` function that accepts a `const void *` and a `const RBNode *`.
1. That first argument has to be castable to a `const ut64 *` for this case, since `_fcn_addr_tree_find_addr` only accepts addrs.
1. I can't reuse `_fcn_addr_tree_cmp` as the comparator function because it expects a `const RAnalFunction *` as its first argument.
1. I could do another comparator function, say  `_fcn_addr_tree_cmp_addr` but:
    1. That just replaces 1 function with another function.
    1. The first argument needs to be a pointer, and thus will cause unnecessary dereferencing.

So I don't think I should use `r_rbtree_find` here.

Reply:
On the other hand you'd just use the provided API, no need to know the internal details of a RBTree and in case RBTree will ever be touched, you won't need to touch this code at all.

But ok for keeping it as is.

Reply:
Ah the sacrifices made for code that appears fast.

Reply:
It would still be fast. Avoiding dereferences, etc. are "tricks" to improve performances, but IMHO they are things that should be considered only when it was proven the slowness comes from there. I'd always prefer clean code before starting doing micro-improvements.

Reply:
> no need to know the internal details

That's where the documentation is.

Reply:
:D